src/averageAges.js

```diff
@@ -37,8 +43,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  const filteredWomen = withChildren
+    ? people.filter(
+      person => people.some(child => child.mother === person.name)
+      && person.sex === 'f')
+    : people.filter(person => person.sex === 'f');
```

Comment:
Same as in `calculateMenAverageAge`. Use 1 filter and expand condition if it needed.